### PR TITLE
entproto/cmd/protoc-gen-entgrpc: grpc codegen support pb wrapper types

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: entproto/cmd/protoc-gen-entgrpc/converter.go [hunk 1]
```diff
@@ -90,29 +92,67 @@ func (g *serviceGenerator) newConverter(fld *entproto.FieldMappingDescriptor) (*
...
+	case fqn == "google.protobuf.Timestamp":
+		conv.toProtoConstructor = protogen.GoImportPath("google.golang.org/protobuf/types/known/timestamppb").Ident("New")
+	case isWrapperType(fqn):
+		typ := strings.Split(fqn, ".")[2]
+		constructor := strings.TrimSuffix(typ, "Value")
+		conv.toProtoConstructor = protogen.GoImportPath("google.golang.org/protobuf/types/known/wrapperspb").Ident(constructor)
+		goType := wrapperPrimitives[fqn]
+		if entFieldType != goType {
+			conv.toProtoConversion = goType
+		}
```
Comment (a8m) [inline entproto/cmd/protoc-gen-entgrpc/converter.go:106]:
```suggestion
		if goType := wrapperPrimitives[fqn]; entFieldType != goType {
			conv.toProtoConversion = goType
		}
```

Cleaner, as it's used only in the if's scope.

File: entproto/cmd/protoc-gen-entgrpc/converter.go [hunk 2]
```diff
@@ -90,29 +92,67 @@ func (g *serviceGenerator) newConverter(fld *entproto.FieldMappingDescriptor) (*
 	}
 	return out, nil
 }
+func convertPbMessageType(fqn, entFieldType string, conv *converter) error {
+	switch {
+	case fqn == "google.protobuf.Timestamp":
+		conv.toProtoConstructor = protogen.GoImportPath("google.golang.org/protobuf/types/known/timestamppb").Ident("New")
+	case isWrapperType(fqn):
```
Comment (a8m) [inline entproto/cmd/protoc-gen-entgrpc/converter.go:99]:
```suggestion
func convertPbMessageType(fd *desc.FieldDescriptor, ft *field.TypeInfo, conv *converter) error {
	switch m := GetMessageType().GetFullyQualifiedName(); {
```

Why passing strings and not the concrete types? You lose the types when you pass strings.

File: entproto/cmd/protoc-gen-entgrpc/converter.go [hunk 3]
```diff
@@ -90,29 +92,67 @@ func (g *serviceGenerator) newConverter(fld *entproto.FieldMappingDescriptor) (*
...
+		}
+		conv.toEntModifier = ".GetValue()"
+	default:
+		return fmt.Errorf("entproto: no mapping for pb field type %q", fqn)
+	}
+	return nil
+}
 
-func (g *serviceGenerator) renderToProto(fc *converter, ident string) string {
+func (g *serviceGenerator) renderToProto(conv *converter, ident string) string {
```
Comment (a8m) [inline entproto/cmd/protoc-gen-entgrpc/converter.go:114]: Why?

**Review by a8m — APPROVED** [review]
Minor comments. Please address them and land it

File: entproto/cmd/protoc-gen-entgrpc/converter.go (hunk 3 above)
Comment (rotemtam) [reply entproto/cmd/protoc-gen-entgrpc/converter.go:114]: fc used to be "fieldCodec" and i missed changing the parameter name

File: entproto/cmd/protoc-gen-entgrpc/converter.go (hunk 2 above)
Comment (rotemtam) [reply entproto/cmd/protoc-gen-entgrpc/converter.go:99]: good point

File: entproto/cmd/protoc-gen-entgrpc/converter.go (hunk 1 above)
Comment (rotemtam) [reply entproto/cmd/protoc-gen-entgrpc/converter.go:106]: <3

**Pull request merged**